frame/merkle-mountain-range/src/mmr/mmr.rs

```diff
@@ -48,6 +48,36 @@ where
 		.map_err(|e| Error::Verify.log_debug(e))
 }
 
+/// Stateless verification of the proof for a batch of leaves.
+/// Note, the leaves should be sorted such that corresponding leaves and leaf indices have the
+/// same position in both the `leaves` vector and the `leaf_indices` vector contained in the
+/// [primitives::BatchProof]
+pub fn verify_leaves_proof<H, L>(
+	root: H::Output,
+	leaves: Vec<Node<H, L>>,
+	proof: primitives::BatchProof<H::Output>,
+) -> Result<bool, Error>
+where
+	H: sp_runtime::traits::Hash,
+	L: primitives::FullLeaf,
+{
+	let size = NodesUtils::new(proof.leaf_count).size();
+
+	let leaves_and_position = proof
```

Comment:
```suggestion
	let leaves_and_positions = proof
```
or more technically correct
```suggestion
	let leaves_positions_and_data = proof
```